modules/javafx.web/src/main/java/com/sun/javafx/webkit/drt/UIClientImpl.java

```diff
@@ -182,7 +183,30 @@ public boolean runBeforeUnloadConfirmPanel(String message) {
      */
     @Override
     public String[] chooseFile(String initialFileName, boolean multiple, String mimeFilters) {
-        throw new UnsupportedOperationException("Not supported yet");
+        if (!DumpRenderTree.drt.complete()) {
+            DumpRenderTree.out.printf("OPEN FILE PANEL\n");
+        }
+        String[] openPanelFiles = DumpRenderTree.drt.openPanelFiles();
+        if (openPanelFiles == null || openPanelFiles.length == 0) {
+            return null;
+        }
+
+        File testURLFile = new File(DumpRenderTree.drt.testURL());
+        String testURLFileParent = testURLFile.getParent();
+        int idx = 0;
+        if (multiple) {
+            String[] result = new String[openPanelFiles.length];
+            for (String str: openPanelFiles) {
+                result[idx++] = testURLFileParent + "\\" + str;
```

Comment:
It may not work on non windows platforms. You should make use of File.separator or Use `new File(parent, child).getAbsolutePath()`